src/browser/modules/Sidebar/ProjectsFilesScripts.tsx

```diff
@@ -0,0 +1,213 @@
+  onExportScripts: () => void
+  onUpdateFolder: () => void
+  onRemoveFolder: () => void
+  scriptsNamespace: string
+  title: string
+}
+
+function ProjectFilesScripts(props: IProjectFilesScripts): JSX.Element {
+  const { data, error: getProjectFilesError, refetch } = useQuery<
+    IProjectFilesResult,
+    IProjectFilesVariables
+  >(GET_PROJECT_FILES, {
+    variables: ProjectFilesQueryVars
+  })
+  const [removeFile, { error: removeProjectFileError }] = useMutation(
+    DELETE_PROJECT_FILE
+  )
+  const [projectFiles, setProjectFiles] = useState<IFavorite[]>([])
+
+  useEffect(() => {
+    let isStillMounted = true
+    if (data) {
+      const getProjectFiles = async () => {
+        const getProjectFilePromises = data.getProject.files.map(
+          mapProjectFileToFavorites
+        )
+        const projectFiles = await Promise.all(getProjectFilePromises)
+        if (isStillMounted) {
+          setProjectFiles(projectFiles)
+        }
+      }
+      getProjectFiles()
+    }
+    return () => {
+      isStillMounted = false
+    }
+  }, [data])
+
+  useEffect(() => {
+    if (data && refetch) {
+      refetch()
+    }
+  }, [data, refetch])
+
+  const myScriptsProps = {
+    ...omit(props, 'bus'),
+    scripts: projectFiles,
+    isProjectFiles: true,
+    onRemoveScript: async (favorite: IFavorite) => {
+      const directory = favorite.path.substring(1) // @todo: adding in SLASH to path
+      const filePath = path.join(directory, favorite.name)
+      try {
+        const { data } = await removeFile({
+          variables: ProjectFileMutationVars(filePath),
+          update: updateCacheRemoveProjectFile
+        })
+        props.bus.send(REMOVE_PROJECT_FILE, {
+          name: data.removeProjectFile.name,
+          directory: data.removeProjectFile.directory,
+          extension: data.removeProjectFile.extension
+        })
+      } catch (e) {
+        console.log(e)
+      }
+    }
+  }
+  return (
+    <>
+      <MyScripts {...myScriptsProps} />
+      <ProjectFilesError
+        apolloErrors={[getProjectFilesError, removeProjectFileError]}
+      />
+    </>
+  )
+}
+
+const mapFavoritesStateToProps = () => {
+  return {
+    scriptsNamespace: SLASH,
+    title: 'Project Files'
+  }
+}
+
+const mapFavoritesDispatchToProps = (
+  dispatch: any,
+  ownProps: { bus: Bus }
+) => ({
+  onSelectScript: (favorite: IFavorite) => {
+    ownProps.bus.send(
+      editor.EDIT_CONTENT,
+      editor.editContent(favorite.id, favorite.contents, true)
+    )
+    ownProps.bus.send(SELECT_PROJECT_FILE, {
+      name: favorite.name,
+      directory: favorite.directory,
+      extension: CYPHER_FILE_EXTENSION
+    })
+  },
+  bus: ownProps.bus,
+  onExecScript: (favorite: IFavorite) =>
+    dispatch(executeCommand(favorite.contents)),
+  onExportScripts: Function.prototype,
+  onUpdateFolder: Function.prototype,
+  onRemoveFolder: Function.prototype
+})
+
+const mergeProps = (stateProps: any, dispatchProps: any) => {
```

Comment:
Is this not the default behaviour?

Reply:
not sure, just a remnant from the previous component I worked from in Browser. Ill take a look

Reply:
going to leave in for now, not entirely sure why it doesn't work if I remove it

Reply:
you should be able to safely remote this as well as line 220

Reply:
removed now